NodeExistsException occurred when proxy start up with sharding configuration

### Which version of ShardingSphere did you use?

`master` branch https://github.com/apache/shardingsphere/commit/ee85debafcd31df31ff4c39bc2d830f0c326e270

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?

`ShardingSphere-Proxy`

```
Exception in thread "main" org.apache.shardingsphere.mode.repository.cluster.ClusterPersistRepositoryException: org.apache.zookeeper.KeeperException$NodeExistsException: KeeperErrorCode = NodeExists for /governance_ds/nodes/compute_nodes/online/proxy/690c4a12-a305-4847-a1cb-8b4a6270ab7b
	at org.apache.shardingsphere.mode.repository.cluster.zookeeper.CuratorZookeeperRepository.persistEphemeral(CuratorZookeeperRepository.java:198)
	at org.apache.shardingsphere.mode.manager.cluster.coordinator.registry.status.compute.service.ComputeNodeStatusService.registerOnline(ComputeNodeStatusService.java:55)
	at org.apache.shardingsphere.mode.manager.cluster.coordinator.RegistryCenter.onlineInstance(RegistryCenter.java:96)
	at org.apache.shardingsphere.mode.manager.cluster.ClusterContextManagerBuilder.registerOnline(ClusterContextManagerBuilder.java:82)
	at org.apache.shardingsphere.mode.manager.cluster.ClusterContextManagerBuilder.build(ClusterContextManagerBuilder.java:57)
	at org.apache.shardingsphere.proxy.initializer.BootstrapInitializer.createContextManager(BootstrapInitializer.java:67)
	at org.apache.shardingsphere.proxy.initializer.BootstrapInitializer.init(BootstrapInitializer.java:57)
	at org.apache.shardingsphere.proxy.Bootstrap.main(Bootstrap.java:52)
Caused by: org.apache.zookeeper.KeeperException$NodeExistsException: KeeperErrorCode = NodeExists for /governance_ds/nodes/compute_nodes/online/proxy/690c4a12-a305-4847-a1cb-8b4a6270ab7b
	at org.apache.zookeeper.KeeperException.create(KeeperException.java:126)
	at org.apache.zookeeper.KeeperException.create(KeeperException.java:54)
	at org.apache.zookeeper.ZooKeeper.create(ZooKeeper.java:1836)
	at org.apache.curator.framework.imps.CreateBuilderImpl$16.call(CreateBuilderImpl.java:1131)
	at org.apache.curator.framework.imps.CreateBuilderImpl$16.call(CreateBuilderImpl.java:1113)
	at org.apache.curator.RetryLoop.callWithRetry(RetryLoop.java:93)
	at org.apache.curator.framework.imps.CreateBuilderImpl.pathInForeground(CreateBuilderImpl.java:1110)
	at org.apache.curator.framework.imps.CreateBuilderImpl.protectedPathInForeground(CreateBuilderImpl.java:593)
	at org.apache.curator.framework.imps.CreateBuilderImpl.forPath(CreateBuilderImpl.java:583)
	at org.apache.curator.framework.imps.CreateBuilderImpl.forPath(CreateBuilderImpl.java:48)
	at org.apache.shardingsphere.mode.repository.cluster.zookeeper.CuratorZookeeperRepository.persistEphemeral(CuratorZookeeperRepository.java:193)
	... 7 more
```

### Configuration
```

- !SHARDING
  auditors:
    sharding_key_required_auditor:
      type: DML_SHARDING_CONDITIONS
  bindingTables:
  - t_order,t_order_item
  defaultAuditStrategy:
    allowHintDisable: true
    auditorNames:
    - sharding_key_required_auditor
  defaultDatabaseStrategy:
    standard:
      shardingAlgorithmName: database_inline
      shardingColumn: user_id
  keyGenerators:
    snowflake:
      type: SNOWFLAKE
  shardingAlgorithms:
    database_inline:
      props:
        algorithm-expression: ds_${user_id % 2}
      type: INLINE
    t_order_inline:
      props:
        algorithm-expression: t_order_${order_id % 2}
      type: INLINE
    t_order_item_inline:
      props:
        algorithm-expression: t_order_item_${order_id % 2}
      type: INLINE
  tables:
    t_order:
      actualDataNodes: ds_${0..1}.t_order_${0..1}
      auditStrategy:
        allowHintDisable: true
        auditorNames:
        - sharding_key_required_auditor
      keyGenerateStrategy:
        column: order_id
        keyGeneratorName: snowflake
      logicTable: t_order
      tableStrategy:
        standard:
          shardingAlgorithmName: t_order_inline
          shardingColumn: order_id
    t_order_item:
      actualDataNodes: ds_${0..1}.t_order_item_${0..1}
      keyGenerateStrategy:
        column: order_item_id
        keyGeneratorName: snowflake
      logicTable: t_order_item
      tableStrategy:
        standard:
          shardingAlgorithmName: t_order_item_inline
          shardingColumn: order_id
```

--- a/shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-core-spring/shardingsphere-jdbc-core-spring-namespace/src/test/java/org/apache/shardingsphere/spring/namespace/fixture/FixtureClusterPersistRepository.java
+++ b/shardingsphere-jdbc/shardingsphere-jdbc-spring/shardingsphere-jdbc-core-spring/shardingsphere-jdbc-core-spring-namespace/src/test/java/org/apache/shardingsphere/spring/namespace/fixture/FixtureClusterPersistRepository.java
@@ -58,6 +58,10 @@ public final class FixtureClusterPersistRepository implements ClusterPersistRepo
     }
     
     @Override
+    public void persistExclusiveEphemeral(final String key, final String value) {
+    }
+    
+    @Override
     public void delete(final String key) {
     }
     

--- a/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/cluster/coordinator/registry/workerid/generator/ClusterWorkerIdGenerator.java
+++ b/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/cluster/coordinator/registry/workerid/generator/ClusterWorkerIdGenerator.java
@@ -80,7 +80,7 @@ public final class ClusterWorkerIdGenerator implements WorkerIdGenerator {
         Long preselectedWorkerId = priorityQueue.poll();
         Preconditions.checkState(null != preselectedWorkerId, "Preselected worker-id can not be null.");
         try {
-            registryCenter.getRepository().persistEphemeral(WorkerIdNode.getWorkerIdGeneratorPath(preselectedWorkerId.toString()), instanceMetaData.getId());
+            registryCenter.getRepository().persistExclusiveEphemeral(WorkerIdNode.getWorkerIdGeneratorPath(preselectedWorkerId.toString()), instanceMetaData.getId());
             return Optional.of(preselectedWorkerId);
         } catch (final ClusterPersistRepositoryException ignore) {
             return Optional.empty();

--- a/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-core/src/test/java/org/apache/shardingsphere/mode/manager/cluster/coordinator/fixture/ClusterPersistRepositoryFixture.java
+++ b/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-core/src/test/java/org/apache/shardingsphere/mode/manager/cluster/coordinator/fixture/ClusterPersistRepositoryFixture.java
@@ -50,6 +50,10 @@ public final class ClusterPersistRepositoryFixture implements ClusterPersistRepo
     }
     
     @Override
+    public void persistExclusiveEphemeral(final String key, final String value) {
+    }
+    
+    @Override
     public void delete(final String key) {
     }
     

--- a/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-core/src/test/java/org/apache/shardingsphere/mode/manager/cluster/coordinator/registry/process/ProcessListClusterPersistRepositoryFixture.java
+++ b/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-core/src/test/java/org/apache/shardingsphere/mode/manager/cluster/coordinator/registry/process/ProcessListClusterPersistRepositoryFixture.java
@@ -56,6 +56,10 @@ public final class ProcessListClusterPersistRepositoryFixture implements Cluster
     }
     
     @Override
+    public void persistExclusiveEphemeral(final String key, final String value) {
+    }
+    
+    @Override
     public void delete(final String key) {
         REGISTRY_DATA.remove(key);
     }

--- a/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-repository/shardingsphere-cluster-mode-repository-api/src/main/java/org/apache/shardingsphere/mode/repository/cluster/ClusterPersistRepository.java
+++ b/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-repository/shardingsphere-cluster-mode-repository-api/src/main/java/org/apache/shardingsphere/mode/repository/cluster/ClusterPersistRepository.java
@@ -42,6 +42,14 @@ public interface ClusterPersistRepository extends PersistRepository {
     void persistEphemeral(String key, String value);
     
     /**
+     * Persist exclusive ephemeral data.
+     *
+     * @param key key of data
+     * @param value is persisted or not
+     */
+    void persistExclusiveEphemeral(String key, String value);
+    
+    /**
      * Persist lock.
      *
      * @param lockKey lock key

--- a/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-repository/shardingsphere-cluster-mode-repository-provider/shardingsphere-cluster-mode-repository-etcd/src/main/java/org/apache/shardingsphere/mode/repository/cluster/etcd/EtcdRepository.java
+++ b/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-repository/shardingsphere-cluster-mode-repository-provider/shardingsphere-cluster-mode-repository-etcd/src/main/java/org/apache/shardingsphere/mode/repository/cluster/etcd/EtcdRepository.java
@@ -106,6 +106,11 @@ public final class EtcdRepository implements ClusterPersistRepository {
     }
     
     @Override
+    public void persistExclusiveEphemeral(final String key, final String value) {
+        persistEphemeral(key, value);
+    }
+    
+    @Override
     public void delete(final String key) {
         client.getKVClient().delete(ByteSequence.from(key, StandardCharsets.UTF_8), DeleteOption.newBuilder().withPrefix(ByteSequence.from(key, StandardCharsets.UTF_8)).build());
     }

--- a/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-repository/shardingsphere-cluster-mode-repository-provider/shardingsphere-cluster-mode-repository-zookeeper-curator/src/main/java/org/apache/shardingsphere/mode/repository/cluster/zookeeper/CuratorZookeeperRepository.java
+++ b/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-repository/shardingsphere-cluster-mode-repository-provider/shardingsphere-cluster-mode-repository-zookeeper-curator/src/main/java/org/apache/shardingsphere/mode/repository/cluster/zookeeper/CuratorZookeeperRepository.java
@@ -40,8 +40,8 @@ import org.apache.shardingsphere.mode.repository.cluster.zookeeper.lock.Zookeepe
 import org.apache.shardingsphere.mode.repository.cluster.zookeeper.props.ZookeeperProperties;
 import org.apache.shardingsphere.mode.repository.cluster.zookeeper.props.ZookeeperPropertyKey;
 import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.KeeperException;
 import org.apache.zookeeper.KeeperException.OperationTimeoutException;
-import org.apache.zookeeper.KeeperException.NodeExistsException;
 import org.apache.zookeeper.ZooDefs;
 import org.apache.zookeeper.data.ACL;
 
@@ -190,11 +190,25 @@ public final class CuratorZookeeperRepository implements ClusterPersistRepositor
     @Override
     public void persistEphemeral(final String key, final String value) {
         try {
+            if (isExisted(key)) {
+                client.delete().deletingChildrenIfNeeded().forPath(key);
+            }
             client.create().creatingParentsIfNeeded().withMode(CreateMode.EPHEMERAL).forPath(key, value.getBytes(StandardCharsets.UTF_8));
             // CHECKSTYLE:OFF
         } catch (final Exception ex) {
             // CHECKSTYLE:ON
-            if (ex instanceof NodeExistsException) {
+            CuratorZookeeperExceptionHandler.handleException(ex);
+        }
+    }
+    
+    @Override
+    public void persistExclusiveEphemeral(final String key, final String value) {
+        try {
+            client.create().creatingParentsIfNeeded().withMode(CreateMode.EPHEMERAL).forPath(key, value.getBytes(StandardCharsets.UTF_8));
+            // CHECKSTYLE:OFF
+        } catch (final Exception ex) {
+            // CHECKSTYLE:ON
+            if (ex instanceof KeeperException.NodeExistsException) {
                 throw new ClusterPersistRepositoryException(ex);
             }
             CuratorZookeeperExceptionHandler.handleException(ex);

--- a/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-repository/shardingsphere-cluster-mode-repository-provider/shardingsphere-cluster-mode-repository-zookeeper-curator/src/main/java/org/apache/shardingsphere/mode/repository/cluster/zookeeper/CuratorZookeeperRepository.java
+++ b/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-repository/shardingsphere-cluster-mode-repository-provider/shardingsphere-cluster-mode-repository-zookeeper-curator/src/main/java/org/apache/shardingsphere/mode/repository/cluster/zookeeper/CuratorZookeeperRepository.java
@@ -41,6 +41,7 @@ import org.apache.shardingsphere.mode.repository.cluster.zookeeper.props.Zookeep
 import org.apache.shardingsphere.mode.repository.cluster.zookeeper.props.ZookeeperPropertyKey;
 import org.apache.zookeeper.CreateMode;
 import org.apache.zookeeper.KeeperException;
+import org.apache.zookeeper.KeeperException.NodeExistsException;
 import org.apache.zookeeper.KeeperException.OperationTimeoutException;
 import org.apache.zookeeper.ZooDefs;
 import org.apache.zookeeper.data.ACL;
@@ -208,7 +209,7 @@ public final class CuratorZookeeperRepository implements ClusterPersistRepositor
             // CHECKSTYLE:OFF
         } catch (final Exception ex) {
             // CHECKSTYLE:ON
-            if (ex instanceof KeeperException.NodeExistsException) {
+            if (ex instanceof NodeExistsException) {
                 throw new ClusterPersistRepositoryException(ex);
             }
             CuratorZookeeperExceptionHandler.handleException(ex);

--- a/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-repository/shardingsphere-cluster-mode-repository-provider/shardingsphere-cluster-mode-repository-zookeeper-curator/src/main/java/org/apache/shardingsphere/mode/repository/cluster/zookeeper/CuratorZookeeperRepository.java
+++ b/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-repository/shardingsphere-cluster-mode-repository-provider/shardingsphere-cluster-mode-repository-zookeeper-curator/src/main/java/org/apache/shardingsphere/mode/repository/cluster/zookeeper/CuratorZookeeperRepository.java
@@ -40,7 +40,6 @@ import org.apache.shardingsphere.mode.repository.cluster.zookeeper.lock.Zookeepe
 import org.apache.shardingsphere.mode.repository.cluster.zookeeper.props.ZookeeperProperties;
 import org.apache.shardingsphere.mode.repository.cluster.zookeeper.props.ZookeeperPropertyKey;
 import org.apache.zookeeper.CreateMode;
-import org.apache.zookeeper.KeeperException;
 import org.apache.zookeeper.KeeperException.NodeExistsException;
 import org.apache.zookeeper.KeeperException.OperationTimeoutException;
 import org.apache.zookeeper.ZooDefs;

--- a/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-repository/shardingsphere-cluster-mode-repository-provider/shardingsphere-cluster-mode-repository-zookeeper-curator/src/test/java/org/apache/shardingsphere/mode/repository/cluster/zookeeper/CuratorZookeeperRepositoryTest.java
+++ b/shardingsphere-mode/shardingsphere-mode-type/shardingsphere-cluster-mode/shardingsphere-cluster-mode-repository/shardingsphere-cluster-mode-repository-provider/shardingsphere-cluster-mode-repository-zookeeper-curator/src/test/java/org/apache/shardingsphere/mode/repository/cluster/zookeeper/CuratorZookeeperRepositoryTest.java
@@ -188,8 +188,10 @@ public final class CuratorZookeeperRepositoryTest {
     @Test
     @SneakyThrows
     public void assertPersistEphemeralExist() {
+        when(existsBuilder.forPath("/test/ephemeral")).thenReturn(new Stat());
         when(protect.withMode(CreateMode.EPHEMERAL)).thenReturn(protect);
         REPOSITORY.persistEphemeral("/test/ephemeral", "value4");
+        verify(backgroundVersionable).forPath("/test/ephemeral");
         verify(protect).forPath("/test/ephemeral", "value4".getBytes(StandardCharsets.UTF_8));
     }
     

--- a/shardingsphere-proxy/shardingsphere-proxy-bootstrap/src/test/java/org/apache/shardingsphere/proxy/fixture/ClusterPersistRepositoryFixture.java
+++ b/shardingsphere-proxy/shardingsphere-proxy-bootstrap/src/test/java/org/apache/shardingsphere/proxy/fixture/ClusterPersistRepositoryFixture.java
@@ -56,6 +56,10 @@ public final class ClusterPersistRepositoryFixture implements ClusterPersistRepo
     }
     
     @Override
+    public void persistExclusiveEphemeral(final String key, final String value) {
+    }
+    
+    @Override
     public void delete(final String key) {
     }
     

--- a/shardingsphere-test/shardingsphere-integration-driver-test/src/test/java/org/apache/shardingsphere/driver/fixture/TestClusterPersistRepository.java
+++ b/shardingsphere-test/shardingsphere-integration-driver-test/src/test/java/org/apache/shardingsphere/driver/fixture/TestClusterPersistRepository.java
@@ -58,6 +58,10 @@ public final class TestClusterPersistRepository implements ClusterPersistReposit
     }
     
     @Override
+    public void persistExclusiveEphemeral(final String key, final String value) {
+    }
+    
+    @Override
     public void delete(final String key) {
     }
     

